apps/studio/src/pages/sites/[siteId]/index.tsx

```diff
@@ -22,12 +26,22 @@ import { CreatePageModal } from "~/features/editing-experience/components/Create
 import { MoveResourceModal } from "~/features/editing-experience/components/MoveResourceModal"
 import { useQueryParse } from "~/hooks/useQueryParse"
 import { type NextPageWithLayout } from "~/lib/types"
+import { buildPermissionsFor } from "~/server/modules/permissions/permissions.util"
 import { AdminCmsSidebarLayout } from "~/templates/layouts/AdminCmsSidebarLayout"
+import { trpc } from "~/utils/trpc"
 
 export const sitePageSchema = z.object({
   siteId: z.coerce.number(),
 })
 
+const getPermissions = (roles: { role: RoleType }[]) => {
+  const builder = new AbilityBuilder<ResourceAbility>(createMongoAbility)
+  roles.forEach(({ role }) => {
+    buildPermissionsFor(role, builder)
+  })
+  return builder.build({ detectSubjectType: () => "Resource" })
+}
+
```

Comment:
from karrui: we might be able to jsut return the `ability` straight from backend and skip all these hassle

Reply:
we can try `pack/unpackRules`, i only tried sending `rules` across but faced difficulties